persist-cli/src/main/java/io/ballerina/persist/components/syntax/SyntaxGenerator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.persist.components.syntax;
+
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.persist.BalException;
+import io.ballerina.persist.components.Client;
+import io.ballerina.persist.components.ClientResource;
+import io.ballerina.persist.components.Function;
+import io.ballerina.persist.models.Entity;
+import io.ballerina.persist.models.Module;
+
+/**
+ * This interface is used to generate the syntax tree for the client.
+ *
+ * @since 0.3.1
+ */
+public interface SyntaxGenerator {
+
+    SyntaxTree getClientSyntax() throws BalException;
+
+    Client generateClient(Module entityModule) throws BalException;
+
+    Function generateInitFunction(Module entityModule);
+
+    ClientResource generateClientResource(Entity entity);
+
+    Function generateCloseFunction();
+
+    Function generatePostFunction(Entity entity);
+
+    Function generatePutFunction(Entity entity);
+
+    Function generateDeleteFunction(Entity entity);
```

Comment:
why we don't have get function here?

Reply:
Common syntax and functions are implemented in the `CommonSyntax`. We don't need a separate method for every client because we only have to change the class name in that syntax.
https://github.com/ballerina-platform/persist-tools/pull/195/files#diff-92396999af3a394f0ca83755e07b2a94abadf489ce5776d8d650274f390cc102R114

Reply:
Updated in [49d2fdc](https://github.com/ballerina-platform/persist-tools/pull/195/commits/49d2fdc3976d5e338fe024ae017b65d48fe5772e)